### PR TITLE
Implement add new address form in side drawer

### PR DESCRIPTION
**Pull request by jplorek-atix**
# Checklist

- [ ] JIRA - \<[LW-6579](https://input-output.atlassian.net/browse/LW-6579)>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

This PR removes the add new address form from right column and adds a side drawer to handle this.
Refactors the AddressDetailDrawer.tsx component to allow saving new addresses.

## Screenshots

Screenshot 2023-05-29 at 15 48 03

Screenshot 2023-05-29 at 15 48 21

Screenshot 2023-05-29 at 15 47 48

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/490/5313965613/index.html) for [55a92633](https://github.com/input-output-hk/lace/pull/57/commits/55a9263331bf4c32c832a4d9347b39588849fa53)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 13     | 6      | 0       | 0     | 19    | ❌     |

[LW-6579]: https://input-output.atlassian.net/browse/LW-6579?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser-extension-wallet/src/hooks/useEnterKeyPress.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ const handleAddressBtn = () => {
   const addressSaveBtn: HTMLButtonElement = document.querySelector('[data-testid="address-form-button-save"]');
   const addressDeleteBtn: HTMLButtonElement = document.querySelector('[data-testid="delete-address-modal-confirm"]');
 
-  if (addressDeleteBtn && !addressDeleteBtn.hasAttribute('disabled')) {
+  if (addressDeleteBtn.offsetHeight && !addressDeleteBtn.hasAttribute('disabled')) {
```
Comment (marcin-mazurek) [inline apps/browser-extension-wallet/src/hooks/useEnterKeyPress.ts:46]: Following up on this discussion: https://github.com/input-output-hk/wallet-world/pull/2729#discussion_r1214226529

Could we then extract it to a const to improve readability:
```
const isDeleteBtnVisible = !!addressDeleteBtn.offsetHeight
```

[the remark this answers is not in the document]
Comment (DominikGuzei) [reply apps/browser-extension-wallet/src/hooks/useEnterKeyPress.ts:46]: To be honest i think the whole `useEnterKeyPress` hook should be refactored as this is just a messy function bound to the button DOM elements of various screens by IDs using query selectors 🙈  but ofc i can make this small adjustment 😆

[the remark this answers is not in the document]
Comment (DominikGuzei) [reply apps/browser-extension-wallet/src/hooks/useEnterKeyPress.ts]: @marcin-mazurek this is improved in my last commit ✅

**Comment by DominikGuzei** [comment]
@marcin-mazurek @jplorek-atix i updated and tested this PR with the latest `main` branch changes 👍

**Review by DominikGuzei — APPROVED** [review]
Updated to latest `main` version and tested locally 👍

**Review by milagi — CHANGES_REQUESTED** [review]
- ‘Save your faorite addresses to find them easily’ not displayed in mockup
- ‘Done’ button instead of ‘Save address’
image

Expected:
image

**Comment by milagi** [comment]
Allan comments (https://input-output.atlassian.net/browse/LW-6579?focusedCommentId=184899)
-Remove description at the top
-'Save address' button instead of 'Done' is OK

Basically the only change that has to be made is removing the top description

**Comment by milagi** [comment]
Another thing to fix: Keyboard shortcuts don't work properly when editing:
-Add address in extended mode
-Refresh page
-Click on saved address
-Press 'Enter' key

**Actual Result:**
Nothing happens

**Expected Result:**
Edit screen is displayed

**Review by bslabiak — APPROVED** [review]

**Pull request merged**